File: 13_NoSQL/examsDbClient_Java/src/main/java/at/spengergasse/model/Teacher.java
package at.spengergasse.model;
import lombok.*;
import org.bson.BsonType;
import org.bson.codecs.pojo.annotations.BsonId;
import org.bson.codecs.pojo.annotations.BsonRepresentation;

import java.math.BigDecimal;
import java.time.LocalTime;
import java.util.ArrayList;
import java.util.HashSet;

@Builder
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Teacher {
    private TeacherName name;
    private Gender gender;
    private Integer hoursPerWeek;
    private TimeOnly lessonsFrom;
    private BigDecimal salary;
    @Builder.Default
    private HashSet<String> homeOfficeDays = new HashSet<String>();
    @Builder.Default
    private ArrayList<Subject> canTeachSubjects = new ArrayList<Subject>();
    @BsonId
    public String getId() {
        return name.getShortname();
    }
}

File: 13_NoSQL/ExamsDb/Model/Teacher.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;

namespace ExamDbGenerator.Model
{

    record Teacher(
        TeacherName Name,
        [property: BsonRepresentation(BsonType.String)] Gender Gender,
        int? HoursPerWeek = null, TimeOnly? LessonsFrom = null,
        // Bei LINQ Berechnungen wie Average, ... würden wir ohne AllowTruncation eine Exception bekommen.
        [property:BsonRepresentation(BsonType.Decimal128, AllowTruncation = true)]
        decimal? Salary = null)
    {
        [BsonId]
        public string Id => Name.Shortname;
        [BsonElement]
        public List<string> HomeOfficeDays { get; set; } = new();
        [BsonElement]
        public List<Subject> CanTeachSubjects { get; set; } = new();
    }

}
